### PR TITLE
Landing

### PR DESCRIPTION
**Pull request by giann**
For review. Don't merge yet: it needs some responsive stuff.

### CODE REVIEW

File: static/lua/snippets.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            _G.print = function(...)
+                printToDOM(output, ...)
+                if (not output.classList:contains("snippet__output--visible")) then
+                    output.className = output.className .. " snippet__output--visible"
+                end
+            end
+
+            output.innerHTML = ""
+
+            if (source.dataset and source.dataset.lang == "lua") then
```
Comment (daurnimator) [inline static/lua/snippets.lua]: No parenthesis required :)

File: static/lua/snippets.lua [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                pcall(load(source.textContent))
```
Comment (daurnimator) [inline static/lua/snippets.lua]: Should check result of load

File: static/lua/web-cli.lua [hunk 3]
```diff
@@ -89,7 +89,7 @@ local function doREPL()
     end
 
     input.value = ""
-    prompt.textContent = _G._PROMPT or "> "
+    -- prompt.textContent = _G._PROMPT or "> "
```
Comment (daurnimator) [inline static/lua/web-cli.lua]: Delete rather than commenting?

File: views/index.etlua [hunk 4]
```diff
@@ -1,19 +1,239 @@
...
-    <img src="/static/images/logo.png" alt="Fengari" class="logo">
-    <h1>Fengari</h1>
-
-    <div class="repl">
-        <code id="fengari-console"></code>
-        <div class="fengari-input-container">
-            <label id="fengari-prompt" for="fengari-input">&gt; </label>
-            <textarea class="lua" id="fengari-input" rows="1"></textarea>
+
+    <div class="section section--header">
```
Comment (daurnimator) [inline views/index.etlua]: Use an actual `<section>`?

File: views/index.etlua [hunk 5]
```diff
@@ -22,3 +242,4 @@
 <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/languages/lua.min.js"></script>
 
 <script src="/static/lua/web-cli.lua" type="text/lua" async></script>
+<script src="/static/lua/snippets.lua" type="text/lua" async></script>
```
Comment (daurnimator) [inline views/index.etlua]: Move over to `application/lua`

File: views/layout.etlua [hunk 6]
```diff
@@ -5,7 +5,7 @@
     <title><%= page_title or "Fengari" %></title>
 
     <link href="https://fonts.googleapis.com/css?family=Libre+Franklin:300,300i,400,400i,600,600i" rel="stylesheet">
-
+
```
Comment (daurnimator) [inline views/layout.etlua]: Random whitespace change?

File: views/index.etlua [hunk 7]
```diff
@@ -1,19 +1,239 @@
...
+            <h2>Test it out !</h2>
+            <div class="repl">
+                <code id="fengari-console"></code>
+                <div class="fengari-input-container">
+                    <label id="fengari-prompt" for="fengari-input">→ </label>
+                    <textarea class="lua" id="fengari-input" rows="1" placeholder="Type some Lua here..."></textarea>
+                </div>
+            </div>
+
+            This REPL is itself <a href="https://github.com/fengari-lua/fengari.io/blob/master/static/lua/web-cli.lua">written in Lua</a>.
```
Comment (daurnimator) [inline views/index.etlua:147]: I don't like links to `master` branch (might die in future). Could we have a build process that writes in the commit id?

**Comment by daurnimator** [comment]
Should mention irc channel.

File: views/index.etlua (hunk 4 above)
Comment (giann) [reply views/index.etlua]: Now i’m confused: what’s the lua way ? If conditions in parenthesis or not ?

Comment (daurnimator) [reply views/index.etlua]: lua way is no parenthesis

File: static/lua/snippets.lua (hunk 2 above)
Comment (giann) [reply static/lua/snippets.lua]: And do what with it ?

Comment (daurnimator) [reply static/lua/snippets.lua]: `console.warn` the error message if nil? (at least so the error message isn't discarded)

File: views/index.etlua [hunk 8]
```diff
@@ -81,10 +81,25 @@ print("This section's title is: " .. document.title)
...
+        assert(coroutine.resume(co))
+    end, delay)
+
+    coroutine.yield(co)
+end
+
 coroutine.wrap(function()
-    local data = fetch("/static/data/test.html")
+    print "Going to sleep now..."
+
```
Comment (daurnimator) [inline views/index.etlua]: Maybe don't have new lines (less vertical space taken by snippets the better)

Comment (giann) [reply views/index.etlua]: I think readability is worth it. Also it does not take that much space: https://imgur.com/a/CkZS4

**Review by daurnimator — APPROVED** [review]

**Pull request merged**